tensorcircuit/translation.py

```diff
@@ -367,30 +369,29 @@ def _translate_qiskit_params(
             if len(p.parameters) == 0:
                 parameters.append(float(p))
                 continue
-            if len(p.parameters) != 1:
-                raise ValueError(
-                    f"Can't translate parameter expression with more than 1 parameters: {p}"
-                )
-            p_real = list(p.parameters)[0]
-            if not isinstance(p_real, ParameterVectorElement):
-                raise TypeError(
-                    "Parameters in parameter expression should be ParameterVectorElement"
-                )
+
             # note "sym" != "sim"
             expr = p.sympify().simplify()
-            # only allow simple expressions like 1.0 * theta
-            if not expr.is_Mul:
-                raise ValueError(f"Unsupported parameter expression: {p}")
-            arg1, arg2 = expr.args
-            if arg1.is_number and arg2.is_symbol:
-                coeff = arg1
-            elif arg1.is_symbol and arg2.is_number:
-                coeff = arg2
-            else:
-                raise ValueError(f"Unsupported parameter expression: {p}")
-            # taking real part here because using complex type will result in a type error
-            # for tf backend when the binding parameter is real
-            parameters.append(float(coeff) * binding_params[p_real.index])
+            if isinstance(expr, symengine.Expr):  # qiskit uses symengine if available
+                expr = expr._sympy_()  # sympy.Expr
+
+            for free_symbol in expr.free_symbols:
+                # replace names: theta[0] -> theta_0
+                # ParameterVector creates symbols with brackets like theta[0]
+                # but sympy.lambdify does not allow brackets in symbol names
+                free_symbol.name = free_symbol.name.replace("[", "_").replace("]", "")
+
+            parameter_list = list(p.parameters)
+            sympy_symbols = [param._symbol_expr for param in parameter_list]
+            # replace names again: theta[0] -> theta_0
+            sympy_symbols = [
+                sympy.Symbol(str(symbol).replace("[", "_").replace("]", ""))
+                for symbol in sympy_symbols
+            ]
+            lam_f = sympy.lambdify(sympy_symbols, expr, modules=backend.name)
```

Comment:
for `modules` arg, there is no pytorch module support for `lambdify`? I think we can stick to `numpy` modules? As I have tested, `modules="numpy"` can support `binding_params` in the tensor format of each backend